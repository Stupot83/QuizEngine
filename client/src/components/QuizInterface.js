import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { getQuizzes } from "../actions/actionsForQuizzes";
import { BrowserRouter as Router, Route, Switch, withRouter } from "react-router-dom";
import Loading from "./Loading";
import QuizList from "./QuizList";
import Quiz from "./Quiz";
import Questions from "./Questions";
import "../sass/QuizInterface.scss";
import Topnav from "./navigation/Topnav";
import Sidenav from "./navigation/Sidenav";
import NotAuthorized from "./NotAuthorized";

class QuizInterface extends Component {
    componentDidMount() {
        this.props.getQuizzes();
    }

    render() {
        const { quizzes, quizzesLoading } = this.props.quizzes;

        let displayContent;

        if (quizzes === null || quizzesLoading) {
            displayContent = <Loading />;
        } else if (quizzes.length > 0) {
            displayContent = (
                <div className="Interface_content_container">
                    <Switch>
                        <Route exact path="/display" quizzes={quizzes} component={QuizList} />
                        <Route exact path="/questions" quizzes={quizzes} component={Questions} />
                        <Route exact path="/quizzes/:quiz" component={Quiz} />
                        <Route component={NotAuthorized} />
                    </Switch>
                </div>
            );
        } else {
            displayContent = (
                <>
                    <div className="Interface_content_container">
                        <Switch>
                            <Route exact path="/display" quizzes={[]} component={QuizList} />
                            <Route exact path="/questions" component={Questions} />
                            <Route component={NotAuthorized} />
                        </Switch>
                    </div>
                </>
            );
        }

        return (
            <Router>
                <Topnav />
                <Sidenav />
                <div className="Interface_container">{displayContent}</div>
            </Router>
        );
    }
}

QuizInterface.propTypes = {
    auth: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
    auth: state.auth,
    quizzes: state.quizzes
});

export default withRouter(connect(mapStateToProps, { getQuizzes })(QuizInterface));
