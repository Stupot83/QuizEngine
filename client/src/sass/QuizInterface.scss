.Interface_container {
 display: flex;
 justify-content: space-between;

 .Interface_content_container {
   display: flex;
   flex-direction: column;
   width: 100%;
 }
}