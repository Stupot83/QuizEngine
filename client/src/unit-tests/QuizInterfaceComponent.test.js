import React from "react";
import { shallow } from "enzyme";
import QuizInterface from "../components/QuizInterface";
import QuizEngineStore from "../quizEngineStore";
import { Provider } from "react-redux";

const mockStore = QuizEngineStore;

describe("QuizInterface component", () => {
    it("should render correctly in debug mode", () => {
        const component = shallow(
            <Provider store={mockStore}>
                <QuizInterface debug />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });

    it("should render correctly", () => {
        const component = shallow(
            <Provider store={mockStore}>
                <QuizInterface />
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });
});
