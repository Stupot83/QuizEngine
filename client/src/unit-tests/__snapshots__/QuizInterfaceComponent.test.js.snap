// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizInterface component should render correctly 1`] = `ShallowWrapper {}`;

exports[`QuizInterface component should render correctly in debug mode 1`] = `ShallowWrapper {}`;
